src/ResourceManager/Sql/Commands.Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/UpdateAzureSqlDatabaseVulnerabilityAssessmentSettings.cs

```diff
@@ -226,28 +226,19 @@ protected override DatabaseVulnerabilityAssessmentSettingsModel ApplyUserInputTo
         /// <param name="model">The model object with the data to be sent to the REST endpoints</param>
         protected override DatabaseVulnerabilityAssessmentSettingsModel PersistChanges(DatabaseVulnerabilityAssessmentSettingsModel model)
         {
-            string storageContainerPath = "";
-            string storageAccountSasKey = "";
-
             if (ParameterSetName == UpdateSettingsWithBlobStorageSasUri)
             {
                 // In this case the user provided a full BlobStorageSasUri (e.g. "https://mystorage.blob.core.windows.net/vulnerability-assessment?st=XXXXXX" )
                 var storageContainerArgs = BlobStorageSasUri.AbsoluteUri.ToString().Split('?');
 
-                storageContainerPath = storageContainerArgs[0];
-                storageAccountSasKey = storageContainerArgs[1];
+                string storageContainerPath = storageContainerArgs[0];
+                string storageAccountSasKey = storageContainerArgs[1];
+				return ModelAdapter.SetDatabaseVulnerabilityAssessmentSettingsWithStorageSasKey(model, storageContainerPath, storageAccountSasKey);
             }
             else 
             {
-                // SetSettingsWithStorageAccountNameParameterSet and SetByInputObjectParameterSet
-                StorageContainerInfo storageContainerInfo = ModelAdapter.CreateBlobStorageContainerAndSAS(model.StorageAccountName,
-                    model.ScanResultsContainerName);
-
-                storageContainerPath = storageContainerInfo.StorageContainerPath;
-                storageAccountSasKey = storageContainerInfo.StorageAccountSasKey;
+				return ModelAdapter.SetDatabaseVulnerabilityAssessmentSettingsWithStorageAccessKey(model);
```

Comment:
How does the behavior change here? Is it exactly the same for users? It seems like the storage container is no longer being created - does it have to already exist?